interpreters/luajit/Makefile

```diff
@@ -0,0 +1,170 @@
+DASM_ARCH = armv7m
+
+MINILUA_O = $(LUAJIT_UNPACK)/src/host/minilua.o
+
+BUILDVM_O = $(LUAJIT_UNPACK)/src/host/buildvm.o $(LUAJIT_UNPACK)/src/host/buildvm_asm.o \
+						$(LUAJIT_UNPACK)/src/host/buildvm_peobj.o $(LUAJIT_UNPACK)/src/host/buildvm_lib.o \
+						$(LUAJIT_UNPACK)/src/host/buildvm_fold.o
+
+HOSTCFLAGS += -DLUAJIT_DISABLE_PROFILE -I$(LUAJIT_UNPACK)/src -I$(LUAJIT_UNPACK)/src/host -D__ARM_ARCH_7M__=1 -DLUAJIT_TARGET=LUAJIT_ARCH_arm -DLJ_ARCH_HASFPU=1 -DLJ_ABI_SOFTFP=0 -DLUAJIT_NO_UNWIND
+HOSTLDFLAGS += -lm
+
+MINILUA_BIN = $(LUAJIT_UNPACK)/src/host/minilua
+BUILDVM_BIN = $(LUAJIT_UNPACK)/src/host/buildvm
+
+LJVM_S= $(LUAJIT_UNPACK)/src/lj_vm.S
+
+LIB_VMDEF= $(LUAJIT_UNPACK)/src/jit/vmdef.lua
+LIB_VMDEFP= $(LIB_VMDEF)
+
+ALL_T= $(MINILUA_BIN) $(MINILUA_BIN)
+ALL_HDRGEN= $(LUAJIT_UNPACK)/src/lj_bcdef.h $(LUAJIT_UNPACK)/src/lj_ffdef.h $(LUAJIT_UNPACK)/src/lj_libdef.h $(LUAJIT_UNPACK)/src/lj_recdef.h $(LUAJIT_UNPACK)/src/lj_folddef.h
+ALL_GEN= $(LJVM_S) $(ALL_HDRGEN) $(LIB_VMDEFP)
+
+HOST_OBJS = $(MINILUA_O) $(BUILDVM_O)
+
+LJLIB_O= $(LUAJIT_UNPACK)/src/lib_base.o $(LUAJIT_UNPACK)/src/lib_math.o $(LUAJIT_UNPACK)/src/lib_bit.o $(LUAJIT_UNPACK)/src/lib_string.o $(LUAJIT_UNPACK)/src/lib_table.o $(LUAJIT_UNPACK)/src/lib_io.o $(LUAJIT_UNPACK)/src/lib_os.o $(LUAJIT_UNPACK)/src/lib_package.o $(LUAJIT_UNPACK)/src/lib_debug.o $(LUAJIT_UNPACK)/src/lib_jit.o $(LUAJIT_UNPACK)/src/lib_ffi.o $(LUAJIT_UNPACK)/src/lib_buffer.o
+
+LJLIB_C= $(LJLIB_O:.o=.c)
+
+.NOTPARALLEL:
+
+$(MINILUA_BIN): $(MINILUA_O)
+	@echo "HOSTLINK  $@"
+	$(Q) $(HOSTCC) -m32 -o $@ $(MINILUA_O) $(HOSTLDFLAGS)
+
+$(LUAJIT_UNPACK)/src/host/buildvm_arch.h: $(DASM_DASC) $(MINILUA_BIN) $(LUAJIT_UNPACK)/src/lj_arch.h $(LUAJIT_UNPACK)/src/lua.h $(LUAJIT_UNPACK)/src/luaconf.h
+	@echo "DYNASM    $@"
+	$(Q)$(MINILUA_BIN) $(LUAJIT_UNPACK)/dynasm/dynasm.lua $(DASM_FLAGS) -o $@ $(DASM_DASC)
+
+$(BUILDVM_BIN): $(BUILDVM_O)
+	@echo "HOSTLINK  $@"
+	$(Q)$(HOSTCC) -m32 -o $@ $(BUILDVM_O)
+
+$(LJVM_S): $(BUILDVM_BIN)
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m elfasm -o $@
+
+$(LUAJIT_UNPACK)/src/lj_bcdef.h: $(BUILDVM_BIN) $(LJLIB_C)
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m bcdef -o $@ $(LJLIB_C)
+
+$(LUAJIT_UNPACK)/src/lj_ffdef.h: $(BUILDVM_BIN) $(LJLIB_C)
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m ffdef -o $@ $(LJLIB_C)
+
+$(LUAJIT_UNPACK)/src/lj_libdef.h: $(BUILDVM_BIN) $(LJLIB_C)
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m libdef -o $@ $(LJLIB_C)
+
+$(LUAJIT_UNPACK)/src/lj_recdef.h: $(BUILDVM_BIN) $(LJLIB_C)
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m recdef -o $@ $(LJLIB_C)
+
+$(LIB_VMDEF): $(BUILDVM_BIN) $(LJLIB_C)
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m vmdef -o $(LIB_VMDEFP) $(LJLIB_C)
+
+$(LUAJIT_UNPACK)/src/lj_folddef.h: $(BUILDVM_BIN) $(LUAJIT_UNPACK)/src/lj_opt_fold.c
+	@echo "BUILDVM   $@"
+	$(Q)$(BUILDVM_BIN) -m folddef -o $@ $(LUAJIT_UNPACK)/src/lj_opt_fold.c
+
+$(HOST_OBJS): %.o: %.c
+	@echo "CC:  $<"
+	$(Q) $(HOSTCC) -m32 -c $(HOSTCFLAGS) $< -o $@
```

Comment:
remove -m32?

Reply:
Why? In LuaJIT documentation is written somewhere, that host application must be built with 32 bit support, when target is 32 bit, otherwise problems. Remember that host application builds buildvm executable, responsible to parse and prepare generated files for the LuaJIT assembler, thus it needs to be on the same page as target is.

"The GNU Makefile-based build system allows cross-compiling on any host for any supported target, as long as both architectures have the same pointer size. If you want to cross-compile to any 32 bit target on an x64 OS, you need to install the multilib development package (e.g. libc6-dev-i386 on Debian/Ubuntu) and build a 32 bit host part (HOST_CC="gcc -m32")."

Reply:
Ok, it's fine if host LuaJIT tool require 32bit